govcd/filter_helpers.go

```diff
@@ -0,0 +1,483 @@
+package govcd
```

Comment:
Maybe naming isn't right, as filter_helpers.go acts as normal library code and as not testing file.
-> filter_helpers_test.go

Reply:
Please read the comments at the top of the file.
If we put it in the `_test` namespace, it can't be used in terraform.
If we put it in a separate package, testing for govcd becomes a circular import.